tests/ui/borrowck/borrowck-struct-update-with-dtor.stderr

```diff
@@ -1,26 +1,191 @@
+   |                           +++++++++++++++++
+
+error[E0509]: cannot move out of type `S<B>`, which implements the `Drop` trait
+  --> $DIR/borrowck-struct-update-with-dtor.rs:24:19
+   |
+LL |         let _s2 = S { a: 2, ..s0 };
+   |                   ^^^^^^^^^^^^^^^^
+   |                   |
+   |                   cannot move out of here
+   |                   move occurs because `s0.b` has type `B`, which does not implement the `Copy` trait
+   |
+note: `B` doesn't implement `Copy` or `Clone`
+  --> $DIR/borrowck-struct-update-with-dtor.rs:4:1
+   |
+LL | struct B;
+   | ^^^^^^^^
+help: if `B` implemented `Clone`, you could clone the value from the field instead of using the spread operator syntax
+   |
+LL |         let _s2 = S { a: 2, b: s0.b.clone(), c: s0.c.clone() };
+   |                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+error[E0509]: cannot move out of type `S<B>`, which implements the `Drop` trait
+  --> $DIR/borrowck-struct-update-with-dtor.rs:24:19
+   |
+LL |         let _s2 = S { a: 2, ..s0 };
+   |                   ^^^^^^^^^^^^^^^^
+   |                   |
+   |                   cannot move out of here
+   |                   move occurs because `s0.c` has type `B`, which does not implement the `Copy` trait
+   |
+note: `B` doesn't implement `Copy` or `Clone`
+  --> $DIR/borrowck-struct-update-with-dtor.rs:4:1
+   |
+LL | struct B;
+   | ^^^^^^^^
+help: if `B` implemented `Clone`, you could clone the value from the field instead of using the spread operator syntax
+   |
+LL |         let _s2 = S { a: 2, b: s0.b.clone(), c: s0.c.clone() };
+   |                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+error[E0509]: cannot move out of type `S<K>`, which implements the `Drop` trait
+  --> $DIR/borrowck-struct-update-with-dtor.rs:29:19
+   |
+LL |         let _s2 = S { a: 2, ..s0 };
+   |                   ^^^^^^^^^^^^^^^^
+   |                   |
+   |                   cannot move out of here
+   |                   move occurs because `s0.b` has type `B`, which does not implement the `Copy` trait
+   |
+note: `B` doesn't implement `Copy` or `Clone`
+  --> $DIR/borrowck-struct-update-with-dtor.rs:4:1
+   |
+LL | struct B;
+   | ^^^^^^^^
+help: if `B` implemented `Clone`, you could clone the value from the field instead of using the spread operator syntax
+   |
+LL |         let _s2 = S { a: 2, b: s0.b.clone(), ..s0 };
+   |                           +++++++++++++++++
+
+error[E0509]: cannot move out of type `S<K>`, which implements the `Drop` trait
+  --> $DIR/borrowck-struct-update-with-dtor.rs:29:19
+   |
+LL |         let _s2 = S { a: 2, ..s0 };
+   |                   ^^^^^^^^^^^^^^^^
+   |                   |
+   |                   cannot move out of here
+   |                   move occurs because `s0.c` has type `K`, which does not implement the `Copy` trait
+   |
+help: clone the value from the field instead of using the spread operator syntax
```

Comment:
is spread operator syntax used anywhere else? I've only seen "functional record update/FRU" or simply "struct update syntax"

Reply:
I think I've gotten confused with it being called that informally in internals, can certainly reword it.